packages/light-client/src/index.ts

```diff
@@ -0,0 +1,473 @@
+ * - During most of period 0, current_sync_committe known, next_sync_committee unknown
+ * - At the end of period 0, get a sync committe update, and populate period 1's committee
+ *
+ * syncCommittees: Map<SyncPeriod, SyncCommittee>, limited to max of 2 items
+ */
+export class Lightclient {
+  readonly api: Api;
+  readonly emitter: LightclientEmitter = mitt();
+
+  readonly config: IBeaconConfig;
+  readonly logger: ILcLogger;
+  readonly genesisValidatorsRoot: Uint8Array;
+  readonly genesisTime: number;
+  readonly beaconApiUrl: string;
+
+  /**
+   * Map of period -> SyncCommittee. Uses a Map instead of spec's current and next fields to allow more flexible sync
+   * strategies. In this case the Lightclient won't attempt to fetch the next SyncCommittee until the end of the
+   * current period. This Map approach is also flexible in case header updates arrive in mixed ordering.
+   */
+  readonly syncCommitteeByPeriod = new Map<SyncPeriod, LightclientUpdateStats & SyncCommitteeFast>();
+  /**
+   * Register participation by period. Lightclient only accepts updates that have sufficient participation compared to
+   * previous updates with a factor of SAFETY_THRESHOLD_FACTOR.
+   */
+  private readonly maxParticipationByPeriod = new Map<SyncPeriod, number>();
+  private head: {
+    participation: number;
+    header: phase0.BeaconBlockHeader;
+    blockRoot: RootHex;
+  };
+
+  private status: RunStatus = {code: RunStatusCode.stopped};
+
+  constructor({config, logger, genesisData, beaconApiUrl, snapshot}: LightclientInitArgs) {
+    this.genesisTime = genesisData.genesisTime;
+    this.genesisValidatorsRoot =
+      typeof genesisData.genesisValidatorsRoot === "string"
+        ? fromHexString(genesisData.genesisValidatorsRoot)
+        : genesisData.genesisValidatorsRoot;
+
+    this.config = createIBeaconConfig(config, this.genesisValidatorsRoot);
+    this.logger = logger ?? getLcLoggerConsole();
+
+    this.beaconApiUrl = beaconApiUrl;
+    this.api = getClient(config, {baseUrl: beaconApiUrl});
+
+    const periodCurr = computeSyncPeriodAtSlot(snapshot.header.slot);
+    const periodNext = periodCurr + 1;
+
+    this.syncCommitteeByPeriod.set(periodCurr, {
+      isFinalized: false,
+      participation: 0,
+      slot: periodCurr * EPOCHS_PER_SYNC_COMMITTEE_PERIOD * SLOTS_PER_EPOCH,
+      ...deserializeSyncCommittee(snapshot.currentSyncCommittee),
+    });
+    this.syncCommitteeByPeriod.set(periodNext, {
+      isFinalized: false,
+      participation: 0,
+      slot: periodNext * EPOCHS_PER_SYNC_COMMITTEE_PERIOD * SLOTS_PER_EPOCH,
+      ...deserializeSyncCommittee(snapshot.nextSyncCommittee),
+    });
+
+    this.head = {
+      participation: 0,
+      header: snapshot.header,
+      blockRoot: toHexString(ssz.phase0.BeaconBlockHeader.hashTreeRoot(snapshot.header)),
+    };
+  }
+
+  static async initializeFromCheckpoint({
+    config,
+    logger,
+    beaconApiUrl,
+    genesisData,
+    checkpoint,
```

Comment:
If we're not using the `checkpoint.epoch` at all (and are really relying on the `checkpoint.block.slot` for timing) we should consider only passing the block root.